### PR TITLE
Enhancement/100 allow switching off sending emails

### LINKED ISSUE #100: As a developer, I would like to do local testing without exceptions due to the "registration email feature"
Whenever I am testing on my local PC and filling in the registration form, I get an exception due to the email server settings not being properly configured. 

We should think about the option of switching off the "send registration emails feature" via a config server setting (or another approach that allows local dev testing without the need of an email server).

### PR DESCRIPTION
**Pull request by gr-hovest-atb**
fixes #100

### CODE REVIEW

File: spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyFetcher.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public ApiKeyFetcher(RestTemplate restTemplate, UrlResourcesProvider urlResource
     }
 
     public List<ApiKey> requestLatestApiKeyListFromAdmin() {
-        log.debug("Requesting latest API key list from CoatRack admin.");
+        log.debug("Requesting latest API key list from " + urlResourcesProvider.getApiKeyListRequestUrl());
```
Comment (BrunoSilvaATB) [inline spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyFetcher.java:56]: This is from another branch or?
It seems that here 3 different issues were joined:
1. The property to allow switching off sending emails
2. The issue about making the Gateway logs better readable for the end users
3. The template to our private profiles
Shouldn't we put those code changes on different MRs?

**Review by BrunoSilvaATB — CHANGES_REQUESTED** [review]
The code regarding allow switching off sending emails seems good.
I just commented about the fact that we are joining different issues here

**Comment by BrunoSilvaATB** [comment]
@gr-hovest-atb I tried to run this MR with maven, however it is still asking me for ygg.mail.server.url (that is required on User Controller)
The weird thing was that I run this with the default profile which contains that url. 
We need to check this, and if this still doesn't work then we need to thing about adding the mail.server.url as an argument too for the run-admin-web-application script
 
image
image
image

File: spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyFetcher.java (hunk 1 above)
Comment (gr-hovest-atb) [reply spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyFetcher.java:56]: To clarify: 
- this has nothing to do with (2.), that is a separate issue. 
- this is built on top of (3.), therefore it is still a draft. After merging master, there will only be the changes from (1.).

Comment (BrunoSilvaATB) [reply spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyFetcher.java]: This was discussed and it is now solved.

**Review by BrunoSilvaATB — APPROVED** [review]

**Review by Monezz — APPROVED** [review]

**Pull request merged**